src/api/course/index.ts

```diff
@@ -0,0 +1,745 @@
+        newCourse.previousPoint.type = `VesselPosition`
+        newCourse.previousPoint.href = null
+      } else {
+        debug(`** Error: navigation.position.value is undefined! (${position})`)
+        return false
+      }
+    } catch (err) {
+      console.log(`** Error: unable to retrieve vessel position!`)
+      return false
+    }
+
+    this.courseInfo = newCourse
+    return true
+  }
+
+  private clearDestination() {
+    this.courseInfo.activeRoute.href = null
+    this.courseInfo.activeRoute.startTime = null
+    this.courseInfo.activeRoute.pointIndex = 0
+    this.courseInfo.activeRoute.pointTotal = 0
+    this.courseInfo.activeRoute.reverse = false
+    this.courseInfo.nextPoint.href = null
+    this.courseInfo.nextPoint.type = null
+    this.courseInfo.nextPoint.position = null
+    this.courseInfo.previousPoint.href = null
+    this.courseInfo.previousPoint.type = null
+    this.courseInfo.previousPoint.position = null
+  }
+
+  private isValidArrivalCircle(value: number | undefined): boolean {
+    return typeof value === 'number' && value >= 0
+  }
+
+  private parsePointIndex(index: number, rte: any): number {
+    if (typeof index !== 'number' || !rte) {
+      return 0
+    }
+    if (!rte.feature?.geometry?.coordinates) {
+      return 0
+    }
+    if (!Array.isArray(rte.feature?.geometry?.coordinates)) {
+      return 0
+    }
+    if (index < 0) {
+      return 0
+    }
+    if (index > rte.feature?.geometry?.coordinates.length - 1) {
+      return rte.feature?.geometry?.coordinates.length - 1
+    }
+    return index
+  }
+
+  private parseHref(href: string): { type: string; id: string } | undefined {
+    if (!href) {
+      return undefined
+    }
+
+    const ref: string[] = href.split('/').slice(-3)
+    if (ref.length < 3) {
+      return undefined
+    }
+    if (ref[0] !== 'resources') {
+      return undefined
+    }
+    return {
+      type: ref[1],
+      id: ref[2]
+    }
+  }
+
+  private getRoutePoint(rte: any, index: number, reverse: boolean) {
+    const pos = reverse
+      ? rte.feature.geometry.coordinates[
+          rte.feature.geometry.coordinates.length - (index + 1)
+        ]
+      : rte.feature.geometry.coordinates[index]
+    return {
+      latitude: pos[1],
+      longitude: pos[0],
+      altitude: pos.length === 3 ? pos[2] : 0
+    }
+  }
+
+  private async getRoute(href: string): Promise<any> {
+    const h = this.parseHref(href)
+    if (h) {
+      try {
+        return await this.server.resourcesApi.getResource(h.type, h.id)
+      } catch (err) {
+        debug(`** Unable to fetch resource: ${h.type}, ${h.id}`)
+        return undefined
+      }
+    } else {
+      debug(`** Unable to parse href: ${href}`)
+      return undefined
+    }
+  }
+
+  private buildDeltaMsg(): any {
+    const values: Array<{ path: string; value: any }> = []
+    const navPath = [
+      'navigation.courseGreatCircle',
+      'navigation.courseRhumbline'
```

Comment:
These are really unfortunate and cause a lot of clutter. The stupid part is that greatcircle / rhumbline has nothing to do with the data we are sending here. Maybe we should consider deprecating these and only use `navigation.course`?

Reply:
I think they have merit for the "calculated" values but not the static values this PR is managing.
So maybe  `activeRoute`, `nextPoint` and `previousPoint` should be removed from `courseGreatCircle` 
 & `navigation.courseRhumbline`.

Reply:
Yes, this was my thinking also.

Reply:
Have created Issue [#636](https://github.com/SignalK/specification/issues/636)